src/handler/http/request/alerts/destinations.rs

```diff
@@ -15,10 +15,24 @@
 
 use std::io::Error;
 
-use actix_web::{delete, get, http, post, put, web, HttpRequest, HttpResponse};
-use config::meta::alerts::destinations::Destination;
+use actix_web::{delete, get, post, put, web, HttpRequest, HttpResponse};
 
-use crate::{common::meta::http::HttpResponse as MetaHttpResponse, service::alerts::destinations};
+use crate::{
+    common::meta::http::HttpResponse as MetaHttpResponse,
+    handler::http::models::destinations::Destination,
+    service::{alerts::destinations, db::alerts::destinations::DestinationError},
+};
+
+impl From<DestinationError> for HttpResponse {
+    fn from(value: DestinationError) -> Self {
+        match value {
+            DestinationError::InUse(e) => MetaHttpResponse::conflict(e),
+            DestinationError::InfraError(e) => MetaHttpResponse::internal_error(e),
+            DestinationError::NotFound => MetaHttpResponse::not_found(DestinationError::NotFound),
+            other_err => MetaHttpResponse::bad_request(other_err),
```

Comment:
(Unrelated to my above comment)

💡 **For future consideration**

I don't know that we necessarily need to change this here, since in a lot of our routes we just serialize the error and return it as an error message. And admittedly, this can be very helpful for debugging. But this just might be something we want to consider for the future...

I believe that generally this can be a security risk since it can expose information about the inner workings of the app to potential attackers. That being said, most of the app is open-source anyways, so I'm not sure how much we're really exposing here. 🤷 Potentially errors from enterprise code could be leaked through these types of error messages? 

I realize too that most users will likely be running the app within their own networks. So my main concern with this would probably apply to (a) running the app as a cloud service, and (b) potentially exposing errors from enterprise code.

Reply:
One potential option could be to do something like...

```
other_err => {
    #[cfg(debug_assertions)]
    MetaHttpResponse::internal_error(other_err),
    
    #[cfg(not(debug_assertions))]
    MetaHttpResponse::internal_error("Unexpected error"),
}
```

Reply:
Just wanted to sort of plant the seeds for a conversation around this. Feel free to ignore for the purposes of this PR

Reply:
For `MetaHttpResponse::...(impl ToString)`, the required argument needs to implement ToString trait. And my test shows that for `thiserror::Error` derived "errors", the "error message" in `#[error(...)]` is returned.

i.e 
```rs
#[derive(Debug, thiserror::Error)]
pub enum DestinationError {
  #[error("Destination not found")]
  NotFound,
}
```

`MetaHttpResponse::not_found(DestinationError::NotFound)` will return `BadRequest` with error message of `Destination not found`

Reply:
Oh yeah, I should have been more clear that this comment was probably more appropriately directed at 

```
            DestinationError::InfraError(e) => MetaHttpResponse::internal_error(e),
```